services/src/main/java/org/keycloak/authentication/authenticators/broker/IdpReviewProfileAuthenticator.java

```diff
@@ -102,23 +101,13 @@ protected void actionImpl(AuthenticationFlowContext context, SerializedBrokeredI
         EventBuilder event = context.getEvent();
         event.event(EventType.UPDATE_PROFILE);
         MultivaluedMap<String, String> formData = context.getHttpRequest().getDecodedFormParameters();
-        UserProfileValidationResult result = forIdpReview(userCtx, formData, context.getSession()).validate();
-
-        List<FormMessage> errors = Validation.getFormErrorsFromValidation(result);
-
-        if (errors != null && !errors.isEmpty()) {
-            Response challenge = context.form()
-                    .setErrors(errors)
-                    .setAttribute(LoginFormsProvider.UPDATE_PROFILE_CONTEXT_ATTR, userCtx)
-                    .setFormData(formData)
-                    .createUpdateProfilePage();
-            context.challenge(challenge);
-            return;
-        }
+        UserModelDelegate updatedProfile = new UserModelDelegate(null) {
```

Comment:
Line 96 of this class calls `!Validation.validateUserMandatoryFields(realm, userCtx)` still. UserProfileSPI should be called instead to validate profile.

Reply:
Yeah, you're right. I also noticed that but I would avoid changing this here but in the next round of changes once we have this PR merged. It was already there already so I did not touch it.